scheduler/core/eventsqueue/events.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import List
+
+from lucupy.minimodel import Resource, Conditions
+
+from scheduler.core.meta import AbstractDataclass, FrozenAbstractDataclass
+
+
+@dataclass(frozen=True)
+class Interruption(FrozenAbstractDataclass):
+    """
+    Parent class for any interruption in the night that would
+    cause a new schedule to be created.
+    """
+    start: datetime
```

Comment:
Thinking out loud: it seems that `start` should maybe be a time slot and not a `datetime`.

It probably doesn't matter because I think we can convert between the two, but we should make sure we have the functionality to do so.

Reply:
I was thinking the user would chose a time in the night instead of a time slot.

Reply:
Ah... that makes sense if the data is coming from a user or from an OCS service (which will have `datetime` labels).

I'm wondering, though, since the Scheduler works on time slots, if we should translate it into a time slot so when we get to time slot `ts_idx`, we can look to see if there are any events, and if there are, then we can grab them, send them to the Scheduler, and check if they should trigger the generation of a new plan.